### ISSUE TITLE
Cancelling a file revision deletes the original file

### ISSUE THREAD

**Issue by luizbgomide** [pkp/pkp-lib#10660]
**Describe the bug**
When you begin to upload a revision to an existent file, after you upload the file if you cancel the process the original file is deleted.

This is problematic if the editor notices they pick the wrong file and data is lost.

**To Reproduce**
1. Go to Submission Files
2. Click on Upload File
3. Choose "this is a revision of an existent file" and upload a file
4. Click on Cancel

**What application are you using?**
 OJS version 3.3.0.13

**Comment by taslangraham** [comment]
This was [fixed](https://github.com/pkp/pkp-lib/issues/8933) on 3.4, and needs to be backported to stable3.3.
This [fix](https://github.com/pkp/pkp-lib/issues/10705) should be applied as well.

**Comment by Tribunal33** [comment]
@taslangraham can we close this or mark as duplicate?  Looks similar to #10705.

I tested the steps listed in the original comments above.  When I upload a revised file and cancel it, it no longer deletes the original file.

**Comment by taslangraham** [comment]
Hey @Tribunal33 This issue is similar to #10705, but they are not the same. However, it was initially fixed on stable 3.4, as indicated in my comment above, but the fix needs to be back-ported to stable3.3 where the issue still exists. Once the fix is merged into stable-3_3_0  then this issue can be closed.

**Comment by taslangraham** [comment]
Ready for review. @asmecher @Vitaliy-1 Can you take a look?
PKP-LIB: https://github.com/pkp/pkp-lib/pull/10836/files
OJS: https://github.com/pkp/ojs/pull/4628/files
OMP: https://github.com/pkp/omp/pull/1818/files
OPS: https://github.com/pkp/ops/pull/853/files

**Comment by asmecher** [comment]
Added a comment, thanks!

**Comment by asmecher** [comment]
If this turns out to be a lot of work for 3.3.0-x, I would be OK with deciding to just support it on newer releases.

**Comment by taslangraham** [comment]
@asmecher Updated and ready for another review

PKP-LIB: https://github.com/pkp/pkp-lib/pull/10836/files

Submodule updates
OJS: https://github.com/pkp/ojs/pull/4628/files
OMP: https://github.com/pkp/omp/pull/1818/files
OPS: https://github.com/pkp/ops/pull/853/files

**Comment by asmecher** [comment]
These look good, thanks; with these PRs, does this mean the issue will be corrected in 3.3.0-x, 3.4.0-x, and `main`?

**Comment by taslangraham** [comment]
Yes, it would have now been fixed across all 3.

**Comment by asmecher** [comment]
Great! The last thing you'll need to do -- since you're modifying JS in our old framework -- is add the rebuilt JS to the PRs. Use `lib/pkp/tools/buildjs.sh -n` to generate it all, then add the resulting JS updates to the application PRs.

**Comment by taslangraham** [comment]
All updated and ready to go

**Comment by asmecher** [comment]
Thanks -- all merged!
